system/classes/Kohana/HTTP/Header.php

```diff
@@ -811,6 +816,81 @@ public function accepts_language_at_quality($language, $explicit = FALSE)
 	}
 
 	/**
+	 * Parses the `Accept-Language:` HTTP header and returns an array containing
+	 * the language names.
+	 *
+	 * @param   string  $language   charset string to parse
+	 * @return  array
+	 * @since   3.3.8
+	 */
+	protected static function _parse_language_header_as_list($language = NULL)
+	{
+		$languages = [];
+		$language  = explode(',', strtolower($language));
+
+		foreach ($language as $lang)
+		{
+			$matches = [];
+
+			if (preg_match('/([\w-]+)\s*(;.*q.*)?/', $lang, $matches))
+			{
+				$languages[] = $matches[1];
+			}
+		}
+
+		return $languages;
+	}
+
+	/**
+	 * Returns the reordered list of supplied `$languages` using the order
+	 * from the `Accept-Language:` HTTP header.
+	 *
+	 * @param   array   $languages  languages to order
+	 * @param   boolean $explicit
+	 * @return  array
+	 * @since   3.3.8
+	 */
+	protected function _order_languages_as_received(array $languages, $explicit = FALSE)
+	{
+		if ($this->_accept_language_list === NULL)
+		{
+			if ($this->offsetExists('Accept-Language'))
+			{
+				$language_header = strtolower($this->offsetGet('Accept-Language'));
+			}
+			else
+			{
+				$language_header = NULL;
+			}
+
+			$this->_accept_language_list = HTTP_Header::_parse_language_header_as_list($language_header);
+		}
+
+		$new_order = [];
+
+		foreach ($this->_accept_language_list as $accept_language)
+		{
+			foreach ($languages as $key => $language)
+			{
+				if (($explicit AND $accept_language == $language) OR
```

Comment:
Is it possible to use a non-strict comparison here?

Reply:
then its ok or not?

Reply:
I think that you need to use a strict comparison in the following lines: 875, 876

The fact is that a non-strict comparison can be unsafe. Or is it not important in this case?